File: src/agent_showdown/interfaces/game/board.py
from pydantic import BaseModel, ConfigDict

from agent_showdown.interfaces.game.obstacle import Obstacle


class Board(BaseModel):
    """The game world: a grid of cells, some of them obstructed.

    Terrain is fixed for the length of a match, so it rides on the board itself rather than
    arriving as events. That is also what carries it to the browser for free: a `Board` is
    already embedded in `game_started` and in the snapshot.
    """

    model_config = ConfigDict(frozen=True)

    width: int
    height: int
    obstacles: tuple[Obstacle, ...] = ()

File: web_client/src/interfaces/game/board.ts
import type { Obstacle } from "./obstacle.js";

/** The playing field, in cells.
 *
 * Terrain is fixed for a match, so it rides on the board rather than arriving as events — which
 * is also how it reaches us, inside `game_started` and inside the snapshot. It is optional
 * because the demo fixture is a recording made before terrain existed.
 */
export interface Board {
  readonly width: number;
  readonly height: number;
  readonly obstacles?: readonly Obstacle[];
}
